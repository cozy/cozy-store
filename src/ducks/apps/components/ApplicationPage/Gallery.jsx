import React, { Component } from 'react'

import { translate } from 'cozy-ui/react/I18n'
import Modal, { ModalContent } from 'cozy-ui/react/Modal'

export class Gallery extends Component {
  constructor(props) {
    super(props)
    this.state = {
      currentImage: null
    }
    this.onClose = this.onClose.bind(this)
  }

  onClick(imageUrl) {
    this.setState({ currentImage: imageUrl })
  }

  onClose() {
    this.setState({ currentImage: null })
  }

  render() {
    const { t, slug, images } = this.props
    const { currentImage } = this.state
    if (!images) return null
    const imagesParts = images.reduce((acc, image, index) => {
      if (index < 2) {
        if (!acc.big) acc.big = []
        acc.big.push(image)
      } else if (index < 5) {
        if (!acc.small) acc.small = []
        acc.small.push(image)
      }
      return acc
    }, {})
    return (
      <div className="sto-app-images">
        {imagesParts.big && !!imagesParts.big.length && (
          <div className="sto-app-big-images">
            {imagesParts.big.map((image, index) => (
              <img
                className="sto-app-big-image"
                src={image}
                alt={`${slug}-image-${index + 1}`}
                key={`${slug}-image-${index + 1}`}
                tabIndex={0}
                onClick={() => this.onClick(image)}
                onKeyUp={e => (e.keyCode === 13 ? this.onClick(image) : null)}
              />
            ))}
          </div>
        )}
        {imagesParts.small && !!imagesParts.small.length && (
          <div
            className={`sto-app-small-images${
              imagesParts.small.length > 2 ? ' --space-around' : ''
            }`}
          >
            {imagesParts.small.map((image, index) => (
              <img
                className="sto-app-small-image"
                src={image}
                alt={`${slug}-image-${index + 3}`}
                key={`${slug}-image-${index + 3}`}
                tabIndex={0}
                onClick={() => this.onClick(image)}
                onKeyUp={e => (e.keyCode === 13 ? this.onClick(image) : null)}
              />
            ))}
          </div>
        )}
        {currentImage && (
          <Modal
            title={t('app_page.preview')}
            secondaryAction={this.onClose}
            size="xxlarge"
            mobileFullscreen
          >
            <ModalContent>
              <img
                className="sto-app-image-preview"
                src={currentImage}
                alt="image-preview"
              />
            </ModalContent>
          </Modal>
        )}
      </div>
    )
  }
}

export default translate()(Gallery)
