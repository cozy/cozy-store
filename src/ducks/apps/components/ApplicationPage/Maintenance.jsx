import React from 'react'

import { translate } from 'cozy-ui/transpiled/react/I18n'
import ReactMarkdownWrapper from 'ducks/components/ReactMarkdownWrapper'

import maintenanceIllu from 'assets/illustrations/maintenance.svg'

export const Maintenance = ({ t, slug }) => {
  const shortDesc = t(`apps.${slug}.maintenance.short_message`)
  const longDesc = t(`apps.${slug}.maintenance.long_message`)
  return (
    <div className="sto-app-maintenance">
      <div className="sto-app-maintenance-header">
        <img
          className="sto-app-maintenance-header-image"
          src={maintenanceIllu}
        />
        <h3 className="u-title-h3">{t('app_page.interrupted')}</h3>
        <p className="sto-app-maintenance-header-desc">{shortDesc}</p>
      </div>
      <div className="sto-app-maintenance-desc">
        <h3 className="u-title-h3">{t('app_page.maintenance')}</h3>
        <ReactMarkdownWrapper source={longDesc} parseEmoji />
      </div>
    </div>
  )
}

export default translate()(Maintenance)
