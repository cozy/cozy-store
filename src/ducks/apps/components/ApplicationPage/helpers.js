import Intents from 'cozy-interapp'
import minilog from 'cozy-minilog'

const log = minilog('Header')

/**
 * @param {import('cozy-client/types/CozyClient').default} client
 * @param {object} intentData
 * @param {object} app
 */
export const handleIntent = async (client, intentData, app) => {
  const intents = new Intents({ client })
  // `intents.createService` throw an error if executed outside of an Intent iframe or if it is called without intentId.
  if (intentData) {
    try {
      const service = await intents.createService()

      if (intentData.data.terminateIfInstalled) {
        return service.terminate(app)
      }
    } catch (error) {
      log.error('Error on openConnector', error)
    }
  }

  return intents.redirect('io.cozy.accounts', {
    konnector: app.slug
  })
}
