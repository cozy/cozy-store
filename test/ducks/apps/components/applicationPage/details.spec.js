'use strict'

/* eslint-env jest */

import React from 'react'
import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import { tMock } from '../../../../jestLib/I18n'
import { Details } from 'ducks/apps/components/ApplicationPage/Details'

import mockApp from '../../_mockPhotosRegistryVersion'
import mockKonnector from '../../_mockPKonnectorTrinlaneRegistryVersion'

Enzyme.configure({ adapter: new Adapter() })

const appManifest = mockApp.manifest
const konnectorManifest = mockKonnector.manifest

const getAppProps = () => {
  return {
    t: tMock,
    description: appManifest.locales.en.long_description,
    changes: appManifest.locales.en.changes,
    categories: appManifest.categories,
    langs: appManifest.langs,
    version: '0.1.0-dev123',
    mobileApps: [
      { type: 'ios', url: '' },
      { type: 'android', url: 'https://mock.app' }
    ],
    developer: appManifest.developer
  }
}

const getKonnectorProps = () => {
  return {
    t: tMock,
    description: konnectorManifest.locales.en.long_description,
    changes: konnectorManifest.locales.en.changes,
    categories: konnectorManifest.categories,
    langs: konnectorManifest.langs,
    developer: konnectorManifest.developer
  }
}

describe('ApplicationPage details component', () => {
  it('should be rendered correctly with provided app', () => {
    const component = shallow(<Details {...getAppProps()} />).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should be rendered correctly with provided konnector', () => {
    const component = shallow(<Details {...getKonnectorProps()} />).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should be rendered correctly provided app with no description, no platforms, no langs and no changes no version', () => {
    const appProps = Object.assign({}, getAppProps())
    appProps.description = ''
    appProps.changes = ''
    appProps.langs = []
    appProps.mobileApps = []
    appProps.developer = {}
    appProps.version = ''
    const component = shallow(<Details {...appProps} />).getElement()
    expect(component).toMatchSnapshot()
  })

  // it('should handle correctly `display more` behaviour on description part', () => {
  //   // window.innerWidth is 1024 by default using jest
  //   const props = getAppProps()
  //   props.description = props.description + props.description + props.description + props.description + props.description + props.description
  //   const component = shallow(<Details {...props} />)
  //   component
  //     .find('.sto-app-description .sto-details-display-more')
  //     .simulate('click')
  //   expect(component.getElement()).toMatchSnapshot()
  // })
  //
  // it('should handle correctly `display more` behaviour on changes part', () => {
  //   // window.innerWidth is 1024 by default using jest
  //   const props = getAppProps()
  //   props.changes = props.changes + props.changes + props.changes + props.changes + props.changes + props.changes
  //   const component = shallow(<Details {...props} />)
  //   component
  //     .find('.sto-app-changes .sto-details-display-more')
  //     .simulate('click')
  //   expect(component.getElement()).toMatchSnapshot()
  // })
})
