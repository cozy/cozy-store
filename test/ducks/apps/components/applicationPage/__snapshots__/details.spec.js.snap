// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationPage details component should be rendered correctly provided app with no description, no platforms, no langs and no changes no version 1`] = `
<div>
  <div
    class="sto-app-details"
  >
    <div
      class="sto-app-descriptions"
    >
      
      
    </div>
    <div
      class="sto-app-additional-details"
    >
      <h3
        class="u-title-h3"
      >
        Informations
      </h3>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Categories
        </div>
        <div
          class="sto-app-info-content"
        >
          The essentials
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Version
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            data-testid="toggleChannels"
          >
            0.1.0
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Languages
        </div>
        <div
          class="sto-app-info-content"
        >
          English, French
        </div>
      </div>
      <div>
        <button
          class="styles__c-btn___3kXsk styles__c-btn--subtle___2rRQ0 styles__c-btn--center___Nny0n sto-app-permissions-button"
          type="submit"
        >
          <span>
            <span>
              Show permissions
            </span>
          </span>
        </button>
      </div>
      <div>
        <h3
          class="u-title-h3"
        >
          Developer Details
        </h3>
        <div
          class="sto-app-developer-infos"
        >
          <span>
            Cozy
          </span>
          <a
            class="sto-app-developer-link"
            href="https://cozy.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://cozy.io
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ApplicationPage details component should be rendered correctly with provided app 1`] = `
<div>
  <div
    class="sto-app-details"
  >
    <div
      class="sto-app-descriptions"
    >
      <div
        class="sto-app-description"
      >
        <h3
          class="u-title-h3"
        >
          Description
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            Features
          </h3>
          <p
            class="md-paragraph"
          >
            Enjoy features to grab your data in you Cozy 
            <span>
              <img
                class="emoji"
                data-codepoints="1f389"
                src="/emoji-data/img-apple-64/1f389.png"
              />
            </span>
          </p>
        </div>
      </div>
      <div
        class="sto-app-changes"
      >
        <h3
          class="u-title-h3"
        >
          What's New?
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            NEW!!!
          </h3>
          <p
            class="md-paragraph"
          >
             Now on the cozy-store :tada
          </p>
        </div>
      </div>
    </div>
    <div
      class="sto-app-additional-details"
    >
      <h3
        class="u-title-h3"
      >
        Informations
      </h3>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Categories
        </div>
        <div
          class="sto-app-info-content"
        >
          The essentials
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Version
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            data-testid="toggleChannels"
          >
            0.1.0
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Languages
        </div>
        <div
          class="sto-app-info-content"
        >
          English, French
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Mobile Application
        </div>
        <div
          class="sto-app-info-content sto-app-info-content--mobile-apps"
        >
          <a
            class="sto-app-info-content-icon"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
          <a
            class="sto-app-info-content-icon"
            href="https://mock.app"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
        </div>
      </div>
      <div>
        <button
          class="styles__c-btn___3kXsk styles__c-btn--subtle___2rRQ0 styles__c-btn--center___Nny0n sto-app-permissions-button"
          type="submit"
        >
          <span>
            <span>
              Show permissions
            </span>
          </span>
        </button>
      </div>
      <div>
        <h3
          class="u-title-h3"
        >
          Developer Details
        </h3>
        <div
          class="sto-app-developer-infos"
        >
          <span>
            Cozy
          </span>
          <a
            class="sto-app-developer-link"
            href="https://cozy.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://cozy.io
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ApplicationPage details component should be rendered correctly with provided konnector 1`] = `
<div>
  <div
    class="sto-app-details"
  >
    <div
      class="sto-app-descriptions"
    />
    <div
      class="sto-app-additional-details"
    >
      <h3
        class="u-title-h3"
      >
        Informations
      </h3>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Categories
        </div>
        <div
          class="sto-app-info-content"
        >
          Transportation
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Version
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            data-testid="toggleChannels"
          >
            Unknown
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Languages
        </div>
        <div
          class="sto-app-info-content"
        >
          French, English
        </div>
      </div>
      <div>
        <button
          class="styles__c-btn___3kXsk styles__c-btn--subtle___2rRQ0 styles__c-btn--center___Nny0n sto-app-permissions-button"
          type="submit"
        >
          <span>
            <span>
              Show permissions
            </span>
          </span>
        </button>
      </div>
      <div>
        <h3
          class="u-title-h3"
        >
          Developer Details
        </h3>
        <div
          class="sto-app-developer-infos"
        >
          <span>
            Cozy
          </span>
          <a
            class="sto-app-developer-link"
            href="https://cozy.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://cozy.io
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ApplicationPage details component should be rendered correctly with provided konnector in Intent 1`] = `
<div>
  <div
    class="sto-app-details"
  >
    <div
      class="sto-app-descriptions"
    />
    <div
      class="sto-app-additional-details"
    >
      <h3
        class="u-title-h3"
      >
        Informations
      </h3>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Categories
        </div>
        <div
          class="sto-app-info-content"
        >
          Transportation
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Version
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            data-testid="toggleChannels"
          >
            Unknown
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Languages
        </div>
        <div
          class="sto-app-info-content"
        >
          French, English
        </div>
      </div>
      <div>
        <button
          class="styles__c-btn___3kXsk styles__c-btn--subtle___2rRQ0 styles__c-btn--center___Nny0n sto-app-permissions-button"
          type="submit"
        >
          <span>
            <span>
              Show permissions
            </span>
          </span>
        </button>
      </div>
      <div>
        <h3
          class="u-title-h3"
        >
          Developer Details
        </h3>
        <div
          class="sto-app-developer-infos"
        >
          <span>
            Cozy
          </span>
          <a
            class="sto-app-developer-link"
            href="https://cozy.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://cozy.io
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ApplicationPage details component should handle channel switching if with installed app from registry provided 1`] = `
<div>
  <div
    class="sto-app-details"
  >
    <div
      class="sto-app-descriptions"
    >
      <div
        class="sto-app-description"
      >
        <h3
          class="u-title-h3"
        >
          Description
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            Features
          </h3>
          <p
            class="md-paragraph"
          >
            Enjoy features to grab your data in you Cozy 
            <span>
              <img
                class="emoji"
                data-codepoints="1f389"
                src="/emoji-data/img-apple-64/1f389.png"
              />
            </span>
          </p>
        </div>
      </div>
      <div
        class="sto-app-changes"
      >
        <h3
          class="u-title-h3"
        >
          What's New?
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            NEW!!!
          </h3>
          <p
            class="md-paragraph"
          >
             Now on the cozy-store :tada
          </p>
        </div>
      </div>
    </div>
    <div
      class="sto-app-additional-details"
    >
      <h3
        class="u-title-h3"
      >
        Informations
      </h3>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Categories
        </div>
        <div
          class="sto-app-info-content"
        >
          The essentials
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Version
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            data-testid="toggleChannels"
          >
            0.1.0
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
        data-testid="betaToggle"
      >
        <div
          class="sto-app-info-header"
        >
          Get Beta
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            class="styles__toggle___3zVsE"
          >
            <input
              class="styles__checkbox___3ko96"
              id="sto-app-undefined-beta-toggle"
              type="checkbox"
            />
            <label
              class="styles__label___3jY1f"
              for="sto-app-undefined-beta-toggle"
            />
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
        data-testid="devToggle"
      >
        <div
          class="sto-app-info-header"
        >
          Get Dev
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            class="styles__toggle___3zVsE"
          >
            <input
              class="styles__checkbox___3ko96"
              id="sto-app-undefined-dev-toggle"
              type="checkbox"
            />
            <label
              class="styles__label___3jY1f"
              for="sto-app-undefined-dev-toggle"
            />
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Languages
        </div>
        <div
          class="sto-app-info-content"
        >
          English, French
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Mobile Application
        </div>
        <div
          class="sto-app-info-content sto-app-info-content--mobile-apps"
        >
          <a
            class="sto-app-info-content-icon"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
          <a
            class="sto-app-info-content-icon"
            href="https://mock.app"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
        </div>
      </div>
      <div>
        <button
          class="styles__c-btn___3kXsk styles__c-btn--subtle___2rRQ0 styles__c-btn--center___Nny0n sto-app-permissions-button"
          type="submit"
        >
          <span>
            <span>
              Show permissions
            </span>
          </span>
        </button>
      </div>
      <div>
        <h3
          class="u-title-h3"
        >
          Developer Details
        </h3>
        <div
          class="sto-app-developer-infos"
        >
          <span>
            Cozy
          </span>
          <a
            class="sto-app-developer-link"
            href="https://cozy.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://cozy.io
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ApplicationPage details component should not suggest Dev channel switching if the context is not in debug mode 1`] = `
<div>
  <div
    class="sto-app-details"
  >
    <div
      class="sto-app-descriptions"
    >
      <div
        class="sto-app-description"
      >
        <h3
          class="u-title-h3"
        >
          Description
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            Features
          </h3>
          <p
            class="md-paragraph"
          >
            Enjoy features to grab your data in you Cozy 
            <span>
              <img
                class="emoji"
                data-codepoints="1f389"
                src="/emoji-data/img-apple-64/1f389.png"
              />
            </span>
          </p>
        </div>
      </div>
      <div
        class="sto-app-changes"
      >
        <h3
          class="u-title-h3"
        >
          What's New?
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            NEW!!!
          </h3>
          <p
            class="md-paragraph"
          >
             Now on the cozy-store :tada
          </p>
        </div>
      </div>
    </div>
    <div
      class="sto-app-additional-details"
    >
      <h3
        class="u-title-h3"
      >
        Informations
      </h3>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Categories
        </div>
        <div
          class="sto-app-info-content"
        >
          The essentials
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Version
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            data-testid="toggleChannels"
          >
            0.1.0
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
        data-testid="betaToggle"
      >
        <div
          class="sto-app-info-header"
        >
          Get Beta
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            class="styles__toggle___3zVsE"
          >
            <input
              class="styles__checkbox___3ko96"
              id="sto-app-undefined-beta-toggle"
              type="checkbox"
            />
            <label
              class="styles__label___3jY1f"
              for="sto-app-undefined-beta-toggle"
            />
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Languages
        </div>
        <div
          class="sto-app-info-content"
        >
          English, French
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Mobile Application
        </div>
        <div
          class="sto-app-info-content sto-app-info-content--mobile-apps"
        >
          <a
            class="sto-app-info-content-icon"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
          <a
            class="sto-app-info-content-icon"
            href="https://mock.app"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
        </div>
      </div>
      <div>
        <button
          class="styles__c-btn___3kXsk styles__c-btn--subtle___2rRQ0 styles__c-btn--center___Nny0n sto-app-permissions-button"
          type="submit"
        >
          <span>
            <span>
              Show permissions
            </span>
          </span>
        </button>
      </div>
      <div>
        <h3
          class="u-title-h3"
        >
          Developer Details
        </h3>
        <div
          class="sto-app-developer-infos"
        >
          <span>
            Cozy
          </span>
          <a
            class="sto-app-developer-link"
            href="https://cozy.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://cozy.io
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ApplicationPage details component should not suggest channel switching if the app is not from registry and not installed 1`] = `
<div>
  <div
    class="sto-app-details"
  >
    <div
      class="sto-app-descriptions"
    >
      <div
        class="sto-app-description"
      >
        <h3
          class="u-title-h3"
        >
          Description
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            Features
          </h3>
          <p
            class="md-paragraph"
          >
            Enjoy features to grab your data in you Cozy 
            <span>
              <img
                class="emoji"
                data-codepoints="1f389"
                src="/emoji-data/img-apple-64/1f389.png"
              />
            </span>
          </p>
        </div>
      </div>
      <div
        class="sto-app-changes"
      >
        <h3
          class="u-title-h3"
        >
          What's New?
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            NEW!!!
          </h3>
          <p
            class="md-paragraph"
          >
             Now on the cozy-store :tada
          </p>
        </div>
      </div>
    </div>
    <div
      class="sto-app-additional-details"
    >
      <h3
        class="u-title-h3"
      >
        Informations
      </h3>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Categories
        </div>
        <div
          class="sto-app-info-content"
        >
          The essentials
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Version
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            data-testid="toggleChannels"
          >
            0.1.0
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Languages
        </div>
        <div
          class="sto-app-info-content"
        >
          English, French
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Mobile Application
        </div>
        <div
          class="sto-app-info-content sto-app-info-content--mobile-apps"
        >
          <a
            class="sto-app-info-content-icon"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
          <a
            class="sto-app-info-content-icon"
            href="https://mock.app"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
        </div>
      </div>
      <div>
        <button
          class="styles__c-btn___3kXsk styles__c-btn--subtle___2rRQ0 styles__c-btn--center___Nny0n sto-app-permissions-button"
          type="submit"
        >
          <span>
            <span>
              Show permissions
            </span>
          </span>
        </button>
      </div>
      <div>
        <h3
          class="u-title-h3"
        >
          Developer Details
        </h3>
        <div
          class="sto-app-developer-infos"
        >
          <span>
            Cozy
          </span>
          <a
            class="sto-app-developer-link"
            href="https://cozy.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://cozy.io
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ApplicationPage details component should not suggest channel switching if the app is not installed 1`] = `
<div>
  <div
    class="sto-app-details"
  >
    <div
      class="sto-app-descriptions"
    >
      <div
        class="sto-app-description"
      >
        <h3
          class="u-title-h3"
        >
          Description
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            Features
          </h3>
          <p
            class="md-paragraph"
          >
            Enjoy features to grab your data in you Cozy 
            <span>
              <img
                class="emoji"
                data-codepoints="1f389"
                src="/emoji-data/img-apple-64/1f389.png"
              />
            </span>
          </p>
        </div>
      </div>
      <div
        class="sto-app-changes"
      >
        <h3
          class="u-title-h3"
        >
          What's New?
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            NEW!!!
          </h3>
          <p
            class="md-paragraph"
          >
             Now on the cozy-store :tada
          </p>
        </div>
      </div>
    </div>
    <div
      class="sto-app-additional-details"
    >
      <h3
        class="u-title-h3"
      >
        Informations
      </h3>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Categories
        </div>
        <div
          class="sto-app-info-content"
        >
          The essentials
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Version
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            data-testid="toggleChannels"
          >
            0.1.0
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Languages
        </div>
        <div
          class="sto-app-info-content"
        >
          English, French
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Mobile Application
        </div>
        <div
          class="sto-app-info-content sto-app-info-content--mobile-apps"
        >
          <a
            class="sto-app-info-content-icon"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
          <a
            class="sto-app-info-content-icon"
            href="https://mock.app"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
        </div>
      </div>
      <div>
        <button
          class="styles__c-btn___3kXsk styles__c-btn--subtle___2rRQ0 styles__c-btn--center___Nny0n sto-app-permissions-button"
          type="submit"
        >
          <span>
            <span>
              Show permissions
            </span>
          </span>
        </button>
      </div>
      <div>
        <h3
          class="u-title-h3"
        >
          Developer Details
        </h3>
        <div
          class="sto-app-developer-infos"
        >
          <span>
            Cozy
          </span>
          <a
            class="sto-app-developer-link"
            href="https://cozy.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://cozy.io
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`ApplicationPage details component should not suggest channel switching if the installed app is not from registry 1`] = `
<div>
  <div
    class="sto-app-details"
  >
    <div
      class="sto-app-descriptions"
    >
      <div
        class="sto-app-description"
      >
        <h3
          class="u-title-h3"
        >
          Description
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            Features
          </h3>
          <p
            class="md-paragraph"
          >
            Enjoy features to grab your data in you Cozy 
            <span>
              <img
                class="emoji"
                data-codepoints="1f389"
                src="/emoji-data/img-apple-64/1f389.png"
              />
            </span>
          </p>
        </div>
      </div>
      <div
        class="sto-app-changes"
      >
        <h3
          class="u-title-h3"
        >
          What's New?
        </h3>
        <div>
          <h3
            class="md-title md-title--h3"
          >
            NEW!!!
          </h3>
          <p
            class="md-paragraph"
          >
             Now on the cozy-store :tada
          </p>
        </div>
      </div>
    </div>
    <div
      class="sto-app-additional-details"
    >
      <h3
        class="u-title-h3"
      >
        Informations
      </h3>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Categories
        </div>
        <div
          class="sto-app-info-content"
        >
          The essentials
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Version
        </div>
        <div
          class="sto-app-info-content"
        >
          <span
            data-testid="toggleChannels"
          >
            0.1.0
          </span>
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Languages
        </div>
        <div
          class="sto-app-info-content"
        >
          English, French
        </div>
      </div>
      <div
        class="sto-app-info"
      >
        <div
          class="sto-app-info-header"
        >
          Mobile Application
        </div>
        <div
          class="sto-app-info-content sto-app-info-content--mobile-apps"
        >
          <a
            class="sto-app-info-content-icon"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
          <a
            class="sto-app-info-content-icon"
            href="https://mock.app"
            rel="noopener noreferrer"
            target="_blank"
          >
            <svg
              class="styles__icon___23x3R"
              height="24px"
              width="24px"
            >
              <use
                xlink:href="#<svg><text>svg icon</text></svg>"
              />
            </svg>
          </a>
        </div>
      </div>
      <div>
        <button
          class="styles__c-btn___3kXsk styles__c-btn--subtle___2rRQ0 styles__c-btn--center___Nny0n sto-app-permissions-button"
          type="submit"
        >
          <span>
            <span>
              Show permissions
            </span>
          </span>
        </button>
      </div>
      <div>
        <h3
          class="u-title-h3"
        >
          Developer Details
        </h3>
        <div
          class="sto-app-developer-infos"
        >
          <span>
            Cozy
          </span>
          <a
            class="sto-app-developer-link"
            href="https://cozy.io"
            rel="noopener noreferrer"
            target="_blank"
          >
            https://cozy.io
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
